CMakeLists.txt

```diff
@@ -284,6 +284,14 @@ mark_as_advanced(
   CMAKE_SHARED_LINKER_FLAGS_FASTBUILD
 )
 
+if(${IREE_ENABLE_RUNTIME_TRACING})
+  if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")  # Selects Clang or GCC.
+    string (APPEND CMAKE_EXE_LINKER_FLAGS " -ldl")
```

Comment:
no globals like this please - they are infectious - and having it be conditional like this means that there's exactly the kind of spooky action at a distance that you are trying to avoid :)

are there any binaries that aren't using iree_cc_binary that need this? I suspect not; so worst case putting this conditional in there for those targets (vs global) is better. Or better yet, don't make it conditional and just always add -ldl to iree_cc_binary targets when the platform supports it - that prevents the "wtf why doesn't it work for me when I change build configurations?" issues.

Reply:
FWIW, here's what we have in Bazel: https://github.com/google/iree/blob/538891d1a9422b8941c0e7622b2e60544d80f39b/iree/build_defs.oss.bzl#L85-L97

(context: https://github.com/google/iree/pull/1897)